packages/agreement-process/test/submitAgreement.test.ts

```diff
@@ -0,0 +1,1742 @@
+      producerId: producer.id,
+    };
+
+    const pendingAgreement = {
+      ...getMockAgreement(),
+      state: agreementState.pending,
+      consumerId: agreement.consumerId,
+      producerId: agreement.producerId,
+      eserviceId: agreement.eserviceId,
+    };
+
+    await addOneAgreement(agreement);
+    await addOneAgreement(pendingAgreement);
+
+    const authData = getRandomAuthData(consumer.id);
+
+    await expect(
+      agreementService.submitAgreement(
+        agreement.id,
+        { consumerNotes: "This is a test" },
+        {
+          authData,
+          correlationId: randomUUID(),
+          serviceName: "AgreementServiceTest",
+          logger: genericLogger,
+        }
+      )
+    ).rejects.toThrowError(
+      agreementAlreadyExists(consumer.id, agreement.eserviceId)
+    );
+  });
+
+  it("should throw a consumerWithNotValidEmail error when consumer has not a contact email", async () => {
+    const consumer = {
+      ...getMockTenant(),
+      mails: [],
+    };
+    const producer = getMockTenant();
+
+    const agreement = {
+      ...getMockAgreement(
+        generateId<EServiceId>(),
+        consumer.id,
+        agreementState.draft
+      ),
+      producerId: producer.id,
+    };
+    await addOneTenant(consumer);
+    await addOneAgreement(agreement);
+
+    const authData = getRandomAuthData(consumer.id);
+
+    await expect(
+      agreementService.submitAgreement(
+        agreement.id,
+        { consumerNotes: "This is a test" },
+        {
+          authData,
+          correlationId: randomUUID(),
+          serviceName: "AgreementServiceTest",
+          logger: genericLogger,
+        }
+      )
+    ).rejects.toThrowError(
+      consumerWithNotValidEmail(agreement.id, consumer.id)
+    );
+  });
+
+  it("should throw an eServiceNotFound error when eservice not exists", async () => {
+    const consumer = {
+      ...getMockTenant(),
+      mails: [
+        {
+          id: generateId(),
+          kind: tenantMailKind.ContactEmail,
+          address: "avalidemailaddressfortenant@testingagreement.com",
+          createdAt: new Date(),
+        },
+      ],
+    };
+    const producer = getMockTenant();
+
+    const agreement = {
+      ...getMockAgreement(),
+      consumerId: consumer.id,
+      producerId: producer.id,
+    };
+
+    await addOneTenant(consumer);
+    await addOneAgreement(agreement);
+
+    const authData = getRandomAuthData(consumer.id);
+
+    await expect(
+      agreementService.submitAgreement(
+        agreement.id,
+        { consumerNotes: "This is a test" },
+        {
+          authData,
+          correlationId: randomUUID(),
+          serviceName: "AgreementServiceTest",
+          logger: genericLogger,
+        }
+      )
+    ).rejects.toThrowError(eServiceNotFound(agreement.eserviceId));
+  });
+
+  it("should throw a notLatestEServiceDescriptor error when eservice doesn't have descriptors with DRAFT state", async () => {
```

Comment:
Actually, if the eservice has draft descriptors, the submission fails: the validation checks that
- the eservice has at least one descriptor NOT in draft state
- the descriptorId in the agreement corresponds to the latest non-draft descriptor (the one with greater version)
- this descriptor is published or suspended

So I would rephrase in:
```suggestion
  it("should throw a notLatestEServiceDescriptor error when eservice has only draft descriptors (no active descriptors)", async () => {
```